### ISSUE TITLE
[REOPENED] [Windows] windowsfilter folder impossible to delete

### ISSUE THREAD

**Issue by Cloudmersive** [moby/moby#41428]
Reopening this issue:

https://github.com/moby/moby/issues/26873

This was never fixed, and is still a problem today in the latest version of Docker

**Comment by thaJeztah** [comment]
I don't think this can be fixed, due to the nature how windowsfilter works, so the `docker-ci-zap.exe` will likely be the solution for cases where you need to clean up that directory (and removing all images/containers using docker itself doesn't work)

**Comment by Cloudmersive** [comment]
@theJeztah but how is it possible that an unfixable bug like this can exist?  Basically Docker for Windows just leaks disk space constantly.  Killing all your containers, running docker system prune -a doesn't actually delete everything.  Steadily it gets bigger, and bigger, and bigger.  Then you have to run this strange zap program to clear it out.

This is clearly just a bug that needs to be fixed...

**Comment by olljanat** [comment]
@Cloudmersive there is so many different use cases and supported configuration combinations on Docker so can plz provide those infos asked by issue template? We will be able to provide much better answers when we understand your use case.

**Comment by Netlims** [comment]
Any update on this...? 
does the docker-ci-zap program means a docker service shutdown? we can't do it in production... also will it delete everything that exists? because that's not the wanted behavior...

we just need to clean the space of unused at all items..

how is this possible to run in production at a lrage scale if a bug like this is present..

**Comment by olljanat** [comment]
@Netlims "docker-ci-zap" is like name says tool made for CI servers. 

It definitely should not be used in production. However you also should not see this issue in production if you treat you servers like cattle instead of pets.

So instead of fighting with issues like this (and others which comes if you keep your Windows container servers too longtime) focus to implement good enough infrastructure as code and re-deploy also production servers at least twice per year (free tip from someone who have spent a way too much time to debug these issues).

**Comment by alonwillmakeit** [comment]
@olljanat Thanks alot for the time and reply... very appreciated
(I work at @Netlims)

Our cause and product is a very very niche and specific type of type and it cannot be allowed to shut down (health care/hospitals) product. for maintenance and take-care routines... 

but i did stumble upon through searching in the linked issues. in
 of one of the tootls that u modified the layer-leak-checker docker thing,

Is it useable still? 

basically our problem is just revovles around windows images (windows server nano/core) which happends to store GB's (70+)
in a 100GB drive, of unused data in essnse... 

so im trying to figure out a solution to this, with taking the circumstances in regard

**Comment by olljanat** [comment]
@alonwillmakeit critical health care applications running in Windows containers? :scream:

https://github.com/lowenna/docker-leak-check should still works as far I'm aware of and I'm not aware any issues caused by it but still, please take backups first because of the fact that it will remove a lot of data and does it on very aggressive way (that code was borrowed from docker-ci-zap)

PS. In case needed you can find me also from Docker community Slack.
